src/discord/index.js

```diff
@@ -164,6 +173,8 @@ export default class Bot extends EventEmitter {
       if (song && this.voiceConnection) {
         const dispatcher = this.voiceConnection.playFile(fp, {volume: 0});
         dispatcher.setVolumeDecibels(BASELINE_DB + song.gain);
+      } else if (this.voiceConnection && this.voiceConnection.dispatcher) {
```

Comment:
so, the branch above this one also relies on `this.voiceConnection` being nonnull

you can probs think of how this can be refactored to simplify this logic